nilearn/glm/second_level/second_level.py

```diff
@@ -1091,24 +1023,55 @@ def non_parametric_inference(
         second_level_input = _sort_input_dataframe(second_level_input)
     sample_map, _ = _process_second_level_input(second_level_input)
 
+    if isinstance(sample_map, SurfaceImage) and smoothing_fwhm is not None:
+        warn(
+            "Parameter 'smoothing_fwhm' is not "
+            "yet supported for surface data.",
+            UserWarning,
+            stacklevel=2,
+        )
+        smoothing_fwhm = None
+
+    if (isinstance(sample_map, SurfaceImage)) and (tfce or threshold):
+        tfce = False
+        threshold = None
+        warn(
+            (
+                "Cluster level inference not yet implemented "
+                "for surface data.\n"
+                f"Setting {tfce=} and {threshold=}."
+            ),
+            UserWarning,
+            stacklevel=2,
+        )
```

Comment:
forces to skip cluster level inference with surface data

Reply:
Yes, can be implemented soonish now.